src/evict/evict_page.c

```diff
@@ -192,6 +200,9 @@ __wt_evict(WT_SESSION_IMPL *session, WT_REF *ref, uint8_t previous_state, uint32
     if (inmem_split)
         goto done;
 
+    /* We're going ahead with eviction. We shouldn't get here if reconciliation isn't allowed. */
+    WT_ASSERT(session, !F_ISSET(session, WT_SESSION_NO_RECONCILE));
```

Comment:
I looked at ```__evict_review()``` and we return ```EBUSY``` for eviction on dirty pages if ```WT_SESSION_NO_RECONCILE``` is set. We allow evicting clean pages with ```WT_SESSION_NO_RECONCILE``` set, as clean pages are just discarded and no reconciliation is needed. We can, in theory, hit this assert for clean pages.

Since we return EBUSY for the cases we care about, I think we do not need this assert anymore. I suggested adding an assert like that but after looking at ```__evict_review``` I think it is not needed,

Reply:
That makes sense.  I'll remove the assert.